cpu/atmega_common/include/cpu.h

```diff
@@ -52,19 +53,23 @@ extern "C" {
 extern volatile uint8_t __in_isr;
 
 /**
- * @brief Flag entering of an ISR
+ * @brief Run this code on entering interrupt routines
  */
-static inline void __enter_isr(void)
+static inline void __attribute__((always_inline)) __enter_isr(void)
 {
     __in_isr = 1;
 }
 
 /**
- * @brief Flag exiting of an ISR
+ * @brief Run this code on exiting interrupt routines
  */
-static inline void __exit_isr(void)
+static inline void __attribute__((always_inline)) __exit_isr(void)
```

Comment:
why always inline?

Reply:
As it will be executed in an ISR, i thought speed could matter.
And if optimization is off it won't 
> GCC does not inline any functions when not optimizing unless you specify the ‘always_inline’ attribute for the function, like this:
> 
> ```
>/* Prototype.  */
> inline void foo (const char) __attribute__((always_inline));
>```

But as the statement is the attribute has to be after the function, i am not sure but, i think it does not matter.

This [benchmark ](https://indico.cern.ch/event/386232/sessions/159923/attachments/771039/1057534/always_inline_performance.pdf) makes me think maybe `inline` only is better, nevertheless i think i forgot to remove it.